tests/php/test-class-amp-pwa-script-sanitizer.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Class AMP_PWA_Script_Sanitizer_Test.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+use AmpProject\AmpWP\ValidationExemption;
+use AmpProject\Dom\Document;
+use AmpProject\Html\Tag;
+use AmpProject\AmpWP\Tests\TestCase;
+
+/**
+ * Tests the PWA Plugin Sanitizer class.
+ *
+ * @coversDefaultClass AMP_PWA_Script_Sanitizer
+ */
+class AMP_PWA_Script_Sanitizer_Test extends TestCase {
+
+	/**
+	 * XPath query to find the scripts element.
+	 *
+	 * @var string
+	 */
+	const SCRIPT_XPATH = '//script[ @id = "wp-navigation-request-properties" or ( @type = "module" and contains( text(), "checkNetworkAndReload()" ) ) ]';
+
+	/**
+	 * Set up.
+	 */
+	public function set_up() {
+		parent::set_up();
+		if (
+			! ( function_exists( 'is_offline' ) && is_offline() ) &&
+			! ( function_exists( 'is_500' ) && is_500() )
+		) {
```

Comment:
Ah, easy fix for coverage problem. This needs to be changed to:

```suggestion
		if ( ! function_exists( 'is_offline' ) || ! function_exists( 'is_500' ) ) {
```